examples/mixed_hmm/model.py

```diff
@@ -0,0 +1,397 @@
+            e_i_dist = plate_g + plate_i + dist.Categorical(
+                **self.params["e_i"]
+            )(value=e_i) * self.raggedness_masks["individual"](t=0)
+
+            log_prob.append(e_i_dist)
+
+            eps_i = (plate_i + plate_g + self.params["eps_i"]["theta"](e_i=e_i))
+
+        elif self.config["individual"]["random"] == "continuous":
+            eps_i = Variable("eps_i", reals(N_state))
+            eps_i_dist = plate_g + plate_i + dist.Normal(**self.params["eps_i"])(value=eps_i)
+
+            log_prob.append(eps_i_dist)
+        else:
+            eps_i = to_funsor(0.)
+
+        # add group-level and individual-level random effects to gamma
+        gamma = gamma + eps_g + eps_i
+
+        N_state = self.config["sizes"]["state"]
+
+        # we've accounted for all effects, now actually compute gamma_y
+        gamma_y = gamma(y_prev="y(t=1)")
+
+        y = Variable("y", bint(N_state))
+        y_dist = plate_g + plate_i + dist.Categorical(
+            probs=gamma_y.exp() / gamma_y.exp().sum()
+        )(value=y)
+
+        # observation 1: step size
+        step_dist = plate_g + plate_i + dist.Gamma(
+            **{k: v(y_curr=y) for k, v in self.params["step"].items()}
+        )(value=self.observations["step"])
+
+        # step size zero-inflation
+        if self.config["zeroinflation"]:
+            step_zi = dist.Categorical(probs=self.params["zi_step"]["zi_param"](y_curr=y))(
+                value="zi_step")
+            step_zi_dist = plate_g + plate_i + dist.Delta(self.config["MISSING"])(
+                value=self.observations["step"])
+            step_dist = (step_zi + Stack("zi_step", (step_dist, step_zi_dist))).reduce(ops.logaddexp, "zi_step")
+
+        # observation 2: step angle
+        angle_dist = plate_g + plate_i + dist.VonMises(
+            **{k: v(y_curr=y) for k, v in self.params["angle"].items()}
+        )(value=self.observations["angle"])
+
+        # observation 3: dive activity
+        omega_dist = plate_g + plate_i + dist.Beta(
+            **{k: v(y_curr=y) for k, v in self.params["omega"].items()}
+        )(value=self.observations["omega"])
+
+        # dive activity zero-inflation
+        if self.config["zeroinflation"]:
+            omega_zi = dist.Categorical(probs=self.params["zi_omega"]["zi_param"](y_curr=y))(
+                value="zi_omega")
+            omega_zi_dist = plate_g + plate_i + dist.Delta(self.config["MISSING"])(
+                value=self.observations["omega"])
+            omega_dist = (omega_zi + Stack("zi_omega", (omega_dist, omega_zi_dist))).reduce(ops.logaddexp, "zi_omega")
+
+        # finally, construct the term for parallel scan reduction
+        hmm_factor = step_dist + angle_dist + omega_dist
+        hmm_factor = hmm_factor * self.raggedness_masks["individual"]
+        hmm_factor = hmm_factor * self.raggedness_masks["timestep"]
+        # copy masking behavior of pyro.infer.TraceEnum_ELBO._compute_model_factors
```

Comment:
Compare to https://github.com/pyro-ppl/pyro/blob/dev/pyro/infer/traceenum_elbo.py#L95